documentation/docker.md

```diff
@@ -0,0 +1,229 @@
+# Docker
+
+## Containers vs Images
+
+It's worth a quick reminder that Containers are created from Images. [Docker's own documentation says](https://www.docker.com/resources/what-container)
+> A Docker container image is a lightweight, standalone, executable package of software that includes everything needed to run an application: code, runtime, system tools, system libraries and settings.
+> Container images become containers at runtime and in the case of Docker containers - images become containers when they run on Docker Engine.
+
+## A multi-stage Dockerfile
+
+- Images created from the `builder` stage are ~357 MB in size
```

Comment:
It may be worth adding a snippet of the builder stage here?

Reply:
Added details of the builder and production stages, along with the Docker blog around the main purposes of multi-stage builds